Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/asyncmsgs/dao/AsyncMsgRecordDao.java

```diff
@@ -656,11 +656,11 @@ private Date calculateExpirationValue(long value, String units) {
         // Convert the long to a Long Object and change the sign to negative so our query value ends up in the past.
         Long longObj = -value;
 
-        if (units.equalsIgnoreCase(NhincConstants.ASYNC_DB_REC_EXP_VAL_UNITS_SEC)) {
+        if (units.equalsIgnoreCase(NhincConstants.ASYNC_DB_REC_EXP_VAL_UNITS_SEC) && units != null) {
```

Comment:
use this instead NhincConstants.ASYNC_DB_REC_EXP_VAL_UNITS_SEC.equalsIgnoreCase(units) and remove units!= null